### PR TITLE
pythonPackages.openai: init at 0.11.4

### PR DESCRIPTION
**Pull request by malob**
###### Motivation for this change

Want to have access to the `openai` Python package in `nixpkgs`.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/openai/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  propagatedBuildInputs = [
+    openpyxl
+    pandas
+    pandas-stubs
+    requests
+    tqdm
+  ];
+
+  # Tests aren't included in package
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/development/python-modules/openai/default.nix:34]: There are tests in the upstream repo. Using the source GitHub could allow to run the tests.

**Review by fabaff — COMMENTED** [review]
New packages need a maintainer.

Could you please add a `pythonImportsCheck`?

**Comment by malob** [comment]
@fabaff, thanks! I've gone ahead and added myself as a maintainer on both packages, and included `pythonImportsCheck`.

File: pkgs/development/python-modules/openai/default.nix (hunk 1 above)
Comment (malob) [reply pkgs/development/python-modules/openai/default.nix:34]: My main concern with using the GitHub source would be that the latest release version there (0.10.5) is older than the version on PyPi (0.11.0).

**Comment by malob** [comment]
Anything else I can do to help get this merged?

File: pkgs/development/python-modules/openai/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  pythonImportsCheck = [ "openai" ];
+
+  meta = with lib; {
+    description = "Python client library for the OpenAI API";
+    homepage = https://github.com/openai/openai-python;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/openai/default.nix:39]:
```suggestion
    homepage = "https://github.com/openai/openai-python";
```

File: pkgs/development/python-modules/openai/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/openai/default.nix:34]: we can fetch https://github.com/openai/openai-python/commit/62f8d40ffc6d3a723f5f1d99fee7febfd47026b5

File: pkgs/development/python-modules/pandas-stubs/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    sha256 = "9f16206e48ce7aaf499703a07e9196e40a62e3364149105fbe1f9a2f82efecf3";
+  };
+
+  propagatedBuildInputs = [
+    typing-extensions
+  ];
+
+  meta = with lib; {
+    description = "Type annotations for Pandas";
+    homepage = https://github.com/VirtusLab/pandas-stubs;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pandas-stubs/default.nix:22]:
```suggestion
    homepage = "https://github.com/VirtusLab/pandas-stubs";
```

File: pkgs/development/python-modules/pandas-stubs/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "9f16206e48ce7aaf499703a07e9196e40a62e3364149105fbe1f9a2f82efecf3";
+  };
+
+  propagatedBuildInputs = [
+    typing-extensions
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pandas-stubs/default.nix:19]: Please add a checkPhase or pythonImportsCheck.

File: pkgs/development/python-modules/openai/default.nix (hunk 1 above)
Comment (malob) [reply pkgs/development/python-modules/openai/default.nix]: Done.

File: pkgs/development/python-modules/pandas-stubs/default.nix (hunk 4 above)
Comment (malob) [reply pkgs/development/python-modules/pandas-stubs/default.nix:26]: Done.

**Comment by malob** [comment]
@SuperSandro2000, I think everything should be good to go with this PR now. Let me know if you'd like any other changes.

File: pkgs/development/python-modules/openai/default.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    owner = "openai";
+    repo = "openai-python";
+    rev = "v${version}";
+    sha256 = "O2O4+GkyMyAxJqMNgiyPKoSXeJk0HGAst02QV6c9mJs=";
+  };
+
+  propagatedBuildInputs = [
+    openpyxl
+    pandas
+    pandas-stubs
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/openai/default.nix:35]: You are sure this is required at runtime?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 147185` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>openai (python39Packages.openai)</li>
<li>python38Packages.openai</li>
<li>python38Packages.pandas-stubs</li>
<li>python39Packages.pandas-stubs</li>
</ul>
</details>

File: pkgs/development/python-modules/openai/default.nix (hunk 5 above)
Comment (malob) [reply pkgs/development/python-modules/openai/default.nix:35]: This is a great question. As far as I understand, mostly from this [comment](https://github.com/openai/openai-python/blob/26fbacb7ff699ed35997010edde3c0aee6599e5b/setup.py#L20) this package is only used with `mypy` to provide type hints. I confess I don’t know much about `mypy` or Python development using Nix, so I’m not sure if this should be a runtime dependency or not.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 147185 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>4 packages built:</summary>  <ul>    <li>openai (python39Packages.openai)</li>    <li>python38Packages.openai</li>    <li>python38Packages.pandas-stubs</li>    <li>python39Packages.pandas-stubs</li>  </ul></details>

**Comment by jonringer** [comment]
@malob sorry for let this drag on forever. We can fix minor issues in follow up PRs

File: pkgs/development/python-modules/openai/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/openai/default.nix:35]: I don't know how we generally handle this.

**Pull request merged**